web/src/server/routes/sessions.ts

```diff
@@ -1230,7 +1230,22 @@ async function requestTerminalSpawn(params: {
       };
     }
 
-    const success = (response.payload as { success?: boolean })?.success === true;
+    // Decode base64 payload if it's a string
+    let payload = response.payload;
+    if (typeof payload === 'string') {
```

Comment:
I wonder if the payload can be anything other than a string?
And why did this work on my machine?

And we kinda need a test case to not regress on this.